internal/admin/admin.go

```diff
@@ -0,0 +1,60 @@
+/*
+ *
+ * Copyright 2021 gRPC authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+// Package admin contains internal implementation for admin service.
+package admin
+
+import "google.golang.org/grpc"
+
+var (
+	// services is a map from name to service register functions.
+	services = make(map[string]func(grpc.ServiceRegistrar) (func(), error))
```

Comment:
Maybe remove the name?  Just make it a slice?

Reply:
I wanted to avoid duplicates, because they panic:
https://github.com/grpc/grpc-go/blob/967933baf52a7bd113bfc23cbf4c5d01b8367d5b/server.go#L596-L598

Reply:
But why would we register the same service multiple times?

Also there's no guarantee that the name used when registering with `admin` is the same as the name used when registering the service with the server.

Reply:
Maybe this should take a list of `grpc.ServiceDesc`, then we can dedup based on the service name.

Reply:
That doesn't work yet, because CSDS's service desc isn't exported.
Removed name. This is internal anyway, should be safe.